hack/dashboard/openshift/deploy-grafana.sh

```diff
@@ -1,56 +1,240 @@
+		grep -Ev 'kubectl.kubernetes.io/last-applied-configuration|{"apiVersion":|creationTimestamp:|resourceVersion:|uid:' >"$BACKUP_CMO_CFG" || {
+		fail "failed to backup $CMO_CM in $MON_NS"
+		return 1
+	}
+
+	ok "Backed up $CMO_CM config to $BACKUP_DIR"
+}
+
+setup_grafana() {
+	header "Setting up Grafana"
+
+	info "Creating Grafana resources"
+	oc_apply_kepler_ns -k hack/dashboard/openshift/grafana-deploy
+
+	wait_until 20 10 "Grafana to be deployed" \
+		oc get grafana,grafanadatasource
+
+	wait_until 10 10 "Grafana CRDs to be Established" \
+		oc wait --for=condition=Established crd grafanas.integreatly.org
+	wait_until 10 10 "Grafana Datasource CRDs to be Established" \
+		oc wait --for=condition=Established crd grafanadatasources.integreatly.org
+	ok "grafana crds created\n"
+
+	info "Creating a grafana instance"
+	oc_apply_kepler_ns -f hack/dashboard/openshift/grafana-config/02-grafana-instance.yaml
+	ok "Grafana created"
+}
+
+config_grafana_sa() {
+	header "Grafana User Project Monitoring Setup"
+
+	oc_apply_kepler_ns -f hack/dashboard/openshift/grafana-config/02-grafana-sa.yaml
+	wait_until 20 2 "Grafana Service Account" \
+		oc_get_kepler_ns serviceaccount $GRAFANA_SA
+
+	run oc adm policy add-cluster-role-to-user cluster-monitoring-view \
+		-z $GRAFANA_SA -n "$KEPLER_DEPLOYMENT_NS"
+
+	ok "grafana $GRAFANA_SA added to cluster-monitoring-view"
+}
+
+setup_grafana_dashboard() {
+	header "Setting up Grafana dashboard"
+
+	info "Creating datasource"
+	local sa_token=""
+	# sa_token="$(oc -n "$KEPLER_DEPLOYMENT_NS" create token "$GRAFANA_SA")"
+	oc -n "$KEPLER_DEPLOYMENT_NS" create token "$GRAFANA_SA" >tmp/grafana-token
+	sa_token=$(cat tmp/grafana-token)
+
+	# Deploy from updated manifest
+	BEARER_TOKEN="$sa_token" \
+		envsubst <hack/dashboard/openshift/grafana-config/03-grafana-datasource-UPDATETHIS.yaml |
+		oc apply -n "$KEPLER_DEPLOYMENT_NS" -f -
+	ok "created grafana datasource \n"
+
+	info "Creating Dashboard"
+	oc_apply_kepler_ns \
+		-f hack/dashboard/openshift/grafana-config/04-grafana-dashboard.yaml
+
+	# NOTE: route name is dependent on the grafana instance
+	wait_until 20 2 "Grafana dashboard" \
+		oc_get_kepler_ns route grafana-route
+
+	ok "created grafana dashboard\n"
+}
+
+show_key_info() {
+	header "Grafana Dashboard Setup Complete"
+
+	local grafana_url=""
+	grafana_url="https://$(oc_get_kepler_ns route grafana-route -o jsonpath='{.spec.host}')/login"
+
+	# disable use find instead of ls
+	# shellcheck disable=SC2012
+	cat <<-EOF
+		  📦 Cluster Monitoring Configuration 
+			    Backup Directory: $BACKUP_DIR
+			$(ls "$BACKUP_DIR" | sed -e "s|^|      • |g")
+
+		  📈 Grafana Configuration:
+
+			   Dashboard URL: $grafana_url
+			           Admin: kepler
+			        Password: kepler
+	EOF
+
+	line 55 heavy
+}
+
+main() {
+	cd "$PROJECT_ROOT"
+	mkdir -p "$BACKUP_DIR"
+
+	validate_cluster || {
+		fail "Cluster validation failed!"
+		info" Fix issues reported above and rerun the script"
```

Comment:
should be space after info.  like `info "Fix issues reported above and rerun the script"`